framework/src/vectorpostprocessors/PointSamplerBase.C

```diff
@@ -161,6 +167,11 @@ PointSamplerBase::getLocalElemContainingPoint(const Point & p)
 {
   const Elem * elem = (*_pl)(p);
 
+  if (elem && _discontinuous_variables)
+    for (const auto s : elem->side_index_range())
+      if (elem->build_side_ptr(s)->contains_point(p))
+        mooseWarning("A PointSampler is used on a face, at ", p, " with elemental variables.");
+
```

Comment:
Since you're going with this - just move this whole section into the if statement below. Only the processor that owns the element needs to check. Also add a break if a side is found that contains the point: there's no need to call the warning multiple times if you're at a node that's contained within multiple faces.

I would also suggest probably adding a parameter to disable this warning. Something like `warn_discontinuous`